table/block_based/full_filter_block.cc

```diff
@@ -261,7 +246,10 @@ void FullFilterBlockReader::MayMatch(
       filter_range.SkipKey(iter);
     }
   }
-  filter_bits_reader->MayMatch(num_keys, &keys[0], &may_match[0]);
+
+  assert(filter_block.GetValue()->filter_bits_reader());
```

Comment:
Consider stuffing `filter_block.GetValue()->filter_bits_reader()` in a variable. And removing this assert, because it's redundant with an if in the same function just 24 lines away. And we immediately deref anyway.